Linkta-org/core
# #225 Implement Firebase Authentication with Password and Email

- **feat: add github signin**
- **add createUserWithEmailAndPassword firebase Auth; Update create user form**
- **add auth with user password and email functionality**


## Description
This PR adds more authentication options with Firebase and Linkta and adds the following: 

- Adds create an Account with username and password on "/userSignup" route
-  Adds sign in capability with email and password on the "/userLogin" route 
- Adds a "sign out" button to the Main Layout header for temporary use
- Adds Sign In with Github functionality. NOTE: this can only be tested once we have the URL for our hosted client app.  


Finish the implementation of using Github Auth to sign into Linkta.  #281  #282
 Related Epic: https://github.com/Linkta-org/core/issues/185


## Type of change



- [x] ✨  New feature (non-breaking change which adds 

## How Can this be tested? Testing Plan to review this PR




Check out this [loom video](https://www.loom.com/share/f1b6139fbf544db0868fe228e799a336?sid=a0cc1a20-203b-446c-b6ac-81d1d8d5cc55) for a walk through of changes.  The console isn't visible in the video, but you'll get the idea. Follow the steps to verify console messages in your testing. 

1. Go to "userSignup" and create an sign up for an account using email and password. You can use the same email you've already signed up with using an [email alias](https://agileway.substack.com/p/email-testing-tip-using-one-email)

## Checklist:



- [x ] I have reviewed the [**Security Best Practices Document**](./docs/SECURITY_BEST_PRACTICES.md).
- [x ] I have ensured that my pull request title is descriptive.

## Additional Information:
